galsim/utilities.py

```diff
@@ -417,3 +417,151 @@ def _gammafn(x):
                0.00000000000000000141, -0.00000000000000000023, 0.00000000000000000002
              )
+
+def interleaveImages(im_list, N, offsets, add_flux=True, suppress_warnings=False,
+    catch_offset_errors=True):
+    """
+    Interleaves two or more images and outputs a larger image.
+
+    The sampling length of simulated images can be set arbitrarily using the `pixel_scale' argument
+    in drawImage() routine appropriately. However, pixel level detector effects can be included
+    only on images drawn at the native pixel scale, which are typically undersampled. Nyquist
+    sampled images that also include the effects of detector non-idealities can be obtained by
+    drawing multiple undersampled images (with the detector effects included) that are offset from
+    each other by a fraction of a pixel. This is equivalent to obtaining a finer sampled image from
+    a dither sequence, except that this routine handles only equispaced offsets, centered at 0. The
+    dither sequence  must be a list of galsim.Image instances supplied through 'im_list'.
+
+    Here's an example script using this routine:
+
+    Interleaving two Gaussian images along the x-axis
+    -------------------------------------------------
+
+        >>> n = 2
+        >>> gal = galsim.Gaussian(sigma=2.8)
+        >>> DX = numpy.arange(0.0,1.0,1./n)
+        >>> DX -= DX.mean()
+        >>> im_list, offsets = [], []
+        >>> for dx in DX:
+            ... offset = galsim.PositionD(dx,0.0)
+            ... offsets.append(offset)
+            ... im = galsim.Image(16,16)
+            ... gal.drawImage(image=im,offset=offset,scale=1.0)
+            ... im.applyNonlinearity(lambda x: x - 0.01*x**2) # detector effects
+            ... im_list.append(im)
+        >>> img = galsim.utilities.interleaveImages(im_list=im_list,N=(n,1),offsets=offsets)
+
+    @param im_list             A list containing the galsim.Image instances to be interleaved.
+    @param N                   Number of images to interleave in either directions. It can be of
+                               type `int' if equal number of images are interleaved in both
+                               directions or a list or tuple of two integers, containing the number
+                               of images in x and y directions respectively.
+    @param offsets             A list containing the offsets as galsim.PositionD instances
+                               corresponding to every image in `im_list'. The offsets must be spaced
+                               equally and must span an entire pixel area. The offset values must
+                               be symmetric around zero, hence taking positive and negative values,
+                               with upper and lower limits of +0.5 and -0.5 (limit values excluded).
+    @param add_flux            Should the routine add the fluxes of all the images (True) or average
+                               them (False)? [default:True]
```

Comment:
our convention is that the colons have spaces after them when declaring defaults (comment applies to this and other places where you have a default statement in this docstring)